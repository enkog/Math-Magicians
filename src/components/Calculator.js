import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>Hello World!</div>
    );
  }
}

export default Calculator;
