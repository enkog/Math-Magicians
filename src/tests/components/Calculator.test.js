import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../../components/UI/Calculator';

describe('Calculator component', () => {
  test('Calculator component renders', () => {
    const component = render(<Calculator />);
    expect(component).toMatchSnapshot();
  });

  test('Calculator renders correctly', () => {
    render(<Calculator />);
    expect(screen.getByRole('table')).toBeInTheDocument();
  });

  test('Addition is carried out correctly', () => {
    const { getByText } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(screen.getAllByText('4')[0]).toBeInTheDocument();
  });

  test('Subtraction is carried out correctly', () => {
    const { getByText } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('-'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(screen.getAllByText('0')[0]).toBeInTheDocument();
  });

  test('Multiplication is carried out correctly', () => {
    const { getByText } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('x'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    expect(screen.getAllByText('6')[0]).toBeInTheDocument();
  });

  test('Division is carried out correctly', () => {
    const { getByText } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('รท'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));
    expect(screen.getAllByText('1')[0]).toBeInTheDocument();
  });
});
