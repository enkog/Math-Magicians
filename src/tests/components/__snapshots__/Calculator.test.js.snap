// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Calculator component renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator-page"
      >
        <h2
          class="calculator-heading"
        >
          Let's do some math!
        </h2>
        <table
          class="calculator table-bordered"
        >
          <tbody>
            <tr
              class="output"
            >
              <td
                colspan="4"
              >
                0
                
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="AC"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="+/-"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="%"
                />
              </td>
              <td
                class="orange-colour"
              >
                <input
                  type="button"
                  value="รท"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="7"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="8"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="9"
                />
              </td>
              <td
                class="orange-colour"
              >
                <input
                  type="button"
                  value="x"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="4"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="5"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="6"
                />
              </td>
              <td
                class="orange-colour"
              >
                <input
                  type="button"
                  value="-"
                />
              </td>
            </tr>
            <tr>
              <td>
                <input
                  type="button"
                  value="1"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="2"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="3"
                />
              </td>
              <td
                class="orange-colour"
              >
                <input
                  type="button"
                  value="+"
                />
              </td>
            </tr>
            <tr>
              <td
                colspan="2"
              >
                <input
                  type="button"
                  value="0"
                />
              </td>
              <td>
                <input
                  type="button"
                  value="."
                />
              </td>
              <td
                class="orange-colour"
              >
                <input
                  type="button"
                  value="="
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-page"
    >
      <h2
        class="calculator-heading"
      >
        Let's do some math!
      </h2>
      <table
        class="calculator table-bordered"
      >
        <tbody>
          <tr
            class="output"
          >
            <td
              colspan="4"
            >
              0
              
            </td>
          </tr>
          <tr>
            <td>
              <input
                type="button"
                value="AC"
              />
            </td>
            <td>
              <input
                type="button"
                value="+/-"
              />
            </td>
            <td>
              <input
                type="button"
                value="%"
              />
            </td>
            <td
              class="orange-colour"
            >
              <input
                type="button"
                value="รท"
              />
            </td>
          </tr>
          <tr>
            <td>
              <input
                type="button"
                value="7"
              />
            </td>
            <td>
              <input
                type="button"
                value="8"
              />
            </td>
            <td>
              <input
                type="button"
                value="9"
              />
            </td>
            <td
              class="orange-colour"
            >
              <input
                type="button"
                value="x"
              />
            </td>
          </tr>
          <tr>
            <td>
              <input
                type="button"
                value="4"
              />
            </td>
            <td>
              <input
                type="button"
                value="5"
              />
            </td>
            <td>
              <input
                type="button"
                value="6"
              />
            </td>
            <td
              class="orange-colour"
            >
              <input
                type="button"
                value="-"
              />
            </td>
          </tr>
          <tr>
            <td>
              <input
                type="button"
                value="1"
              />
            </td>
            <td>
              <input
                type="button"
                value="2"
              />
            </td>
            <td>
              <input
                type="button"
                value="3"
              />
            </td>
            <td
              class="orange-colour"
            >
              <input
                type="button"
                value="+"
              />
            </td>
          </tr>
          <tr>
            <td
              colspan="2"
            >
              <input
                type="button"
                value="0"
              />
            </td>
            <td>
              <input
                type="button"
                value="."
              />
            </td>
            <td
              class="orange-colour"
            >
              <input
                type="button"
                value="="
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
